packages/core/src/docdb/commands/modifyInstance.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, ToolkitError } from '../../shared'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { DBClusterNode } from '../explorer/dbClusterNode'
+import { DBInstanceNode } from '../explorer/dbInstanceNode'
+import { DBCluster, ModifyDBInstanceMessage } from '@aws-sdk/client-docdb'
+import { DBStorageType, DocumentDBClient } from '../../shared/clients/docdbClient'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { isValidResponse } from '../../shared/wizards/wizard'
+import { telemetry } from '../../shared/telemetry'
+
+/**
+ * Modifies a DocumentDB instance.
+ *
+ * Prompts the user for the instance class.
+ * Updates the instance.
+ * Refreshes the node.
+ */
+export async function modifyInstance(node: DBInstanceNode) {
+    getLogger().debug('ModifyInstance called for: %O', node)
+
+    await telemetry.docdb_resizeInstance.run(async () => {
+        if (!node) {
+            throw new ToolkitError('No node specified for ModifyInstance')
+        }
+
+        if (node.instance.DBInstanceStatus !== 'available') {
+            void vscode.window.showErrorMessage(
+                localize('AWS.docdb.modifyInstance.instanceStopped', 'Instance must be running')
+            )
+            throw new ToolkitError('Instance not available', { cancelled: true })
+        }
+
+        const instanceName = node.instance.DBInstanceIdentifier
+        const parent = node.parent as DBClusterNode
```

Comment:
Do we need to cast here? Should we already know that node.parent is a DBClusterNode?

Reply:
removed unnecessary cast